content/Services/RHOBS/design.md

```diff
@@ -0,0 +1,196 @@
+
+> NOTE: Telemeter was the first responsibility of our service team in the past. We build RHOBS in a way that Telemeter can be a single-region hard tenant. It is isolated because it has different characteristics as typical observability data. It contains very cardinal data with only ~5m sample interval. There is no alerting, only analytic PromQL queries and integration with analytics tools like [obslytics](https://github.com/thanos-community/obslytics). We will focus on `MST` in this design as this is what will be used by the majority of the users.
+
+The set of features, topology and interactions are the same across all RHOBS shards. In addition, each shard allows secure and multi-tenant:
+
+* Metric ingestion in Prometheus data model.
+* Metric storage, including long-term storage using object storage.
+* Querying APIs using PromQL language.
+* Other metric,rule and alert retrieval APIs [supported by Prometheus](https://prometheus.io/docs/prometheus/latest/querying/api/): [Full list](https://github.com/observatorium/api/blob/23af39e5f09845a12dec7179c47b0b364f0a35b0/api/metrics/v1/http.go#L27).
+* Custom recording rule evaluation.
+* Custom alerting rule evaluation and routing.
+
+At the current moment, the design assumes no cross-communication across shards. No data replication or cross-shard queries. However, in future, there is a plan to [add a federation layer that will allow cross-shard querying](https://issues.redhat.com/browse/RHOBS-40) and a truly global view of observability data. Currently, the user can do that on their own using software like [Thanos Querier](https://thanos.io/tip/components/query.md/, [promxy](https://github.com/jacksontj/promxy) or by graphing multiple Queries with [Grafana](https://grafana.com/grafana/dashboards/).
+
+The RHOBS have [the following dependencies](https://visual-app-interface.devshift.net/services#/services/rhobs/app.yml):
+
+![](../../assets/rhobs-deps.png)
+
+From the deployment point of view:
+
+* AWS (EC2 and S3)
+* OSD with AppSRE local monitoring.
+* [Red Hat SSO](https://sso.redhat.com/auth/realms/redhat-external)
+* [Account Manager Service (AMS)](https://visual-app-interface.devshift.net/services#/services/ocm/ams/app.yml)
+
+> NOTE: We also have a set of staging and testing clusters, which are not covered in this document.
+
+Now, look at a single shard RHOBS architecture at the microservice level.
+
+### Services and Interaction Points
+
+The ful ![](../../assets/rhobs-microsvc.png)
```

Comment:
```suggestion
![](../../assets/rhobs-microsvc.png)
```

Reply:
I can't really comment on the diagram file, so commenting here instead.

So currently Thanos Stateless Ruler is not used. It is deployed, but is defunct, as we do not configure any Rules on it. After an attempt to deploy it last time it was decided that in its current state it is not able to perform well with evaluating and remote writing Rule data for multiple tenants, which is something we need. 

Also, Thanos Receive `ketama`  hashing is only on testing cluster for now.

If this diagram represents a near-term future RHOBS service then I think the above points can be ignored, as we plan to switch those components! 🙂

Reply:
We also [plan](https://github.com/rhobs/configuration/pull/301) to deploy obsctl-reloader as an acceptance testing component for generating load in rhobs tenant, for Rules API SLI measurement. 🙂

Reply:
> So currently Thanos Stateless Ruler is not used. It is deployed, but is defunct, as we do not configure any Rules on it. After an attempt to deploy it last time it was decided that in its current state it is not able to perform well with evaluating and remote writing Rule data for multiple tenants, which is something we need.

True, but that's a detail. We know the future direction. I can put that into a risk though (: 

> We also https://github.com/rhobs/configuration/pull/301 to deploy obsctl-reloader as an acceptance testing component for generating load in rhobs tenant, for Rules API SLI measurement. slightly_smiling_face

Nice!